docs/operators/transform.md

```diff
@@ -0,0 +1,121 @@
+
+#### Stateless partitioning of TransformOperator
+Stateless partitioning will ensure that TransformOperator will will be partitioned right at the starting of the application and will remain partitioned throughout the lifetime of the DAG.
+TransformOperator can be stateless partitioned by adding following lines to properties.xml:
+
+```xml
+  <property>
+    <name>dt.operator.{OperatorName}.attr.PARTITIONER</name>
+    <value>com.datatorrent.common.partitioner.StatelessPartitioner:3</value>
+  </property>
+```
+
+where {OperatorName} is the name of the TransformOperator operator.
+Above lines will partition TransformOperator statically 3 times. Above value can be changed accordingly to change the number of static partitions.
+
+#### Dynamic Partitioning of TransformOperator
+Dynamic partitioning is a feature of Apex platform which changes the partition of the operator based on certain condition.
+TransformOperator can be dynamically partitioned using the below two partitioners:
+
+##### Throughput based
+Following code can be added to populateDAG method of application to dynamically partitioning TransformOperator:
+```java
+    StatelessThroughputBasedPartitioner<TransformOperator> partitioner = new StatelessThroughputBasedPartitioner<>();
+    partitioner.setCooldownMillis(conf.getLong(COOL_DOWN_MILLIS, 10000));
+    partitioner.setMaximumEvents(conf.getLong(MAX_THROUGHPUT, 30000));
+    partitioner.setMinimumEvents(conf.getLong(MIN_THROUGHPUT, 10000));
+    dag.setAttribute(transform, OperatorContext.STATS_LISTENERS, Arrays.asList(new StatsListener[]{partitioner}));
+    dag.setAttribute(transform, OperatorContext.PARTITIONER, partitioner);
+```
+
+Above code will dynamically partition TransformOperator when the throughput changes.
+If the overall throughput of TransformOperator goes beyond 30000 or less than 10000, the platform will repartition TransformOperator 
+to balance throughput of a single partition to be between 10000 and 30000.
+CooldownMillis of 10000 will be used as the threshold time for which the throughout change is observed.
+
+Source code for this dynamic application can be found [here] (https://github.com/DataTorrent/examples/blob/master/tutorials/transform/src/main/java/com/example/transform/DynamicTransformApplication.java).
+
+##### Latency based
+Following code can be added to populateDAG method of application to dynamically partitioning TransformOperator:
+```java
+    StatelessLatencyBasedPartitioner<TransformOperator> partitioner = new StatelessLatencyBasedPartitioner<>();
+    partitioner.setCooldownMillis(conf.getLong(COOL_DOWN_MILLIS, 10000));
+    partitioner.setMaximumLatency(conf.getLong(MAX_THROUGHPUT, 10));
+    partitioner.setMinimumLatency(conf.getLong(MIN_THROUGHPUT, 3));
+    dag.setAttribute(transform, OperatorContext.STATS_LISTENERS, Arrays.asList(new StatsListener[]{partitioner}));
+    dag.setAttribute(transform, OperatorContext.PARTITIONER, partitioner);
+```
```

Comment:
Please look at the fileOutput example where the PARTITIONER attribute is set from a properties file.